src/repository/default.rs

```diff
@@ -59,6 +59,39 @@ macro_rules! implement_repository {
 #[async_trait]
 impl Repository for $repo {
     async fn setup(&self) {
+        // TODO remove the entire block a couple of releases past 0.5.7
+        {
+            // Migrate the old field JSON format to the canonical one
```

Comment:
I've tested the migration locally, it successfully converted 
```sql
{"children":[],"name":"cc_call_center_sk","nullable":true,"type":{"bitWidth":32,"isSigned":true,"name":"int"}}
{"children":[],"name":"cc_call_center_id","nullable":true,"type":{"name":"utf8"}}
{"children":[],"name":"cc_rec_start_date","nullable":true,"type":{"name":"date","unit":"DAY"}}
...
{"children":[],"name":"RefererHash","nullable":false,"type":{"bitWidth":64,"isSigned":true,"name":"int"}}
{"children":[],"name":"URLHash","nullable":false,"type":{"bitWidth":64,"isSigned":true,"name":"int"}}
{"children":[],"name":"CLID","nullable":false,"type":{"bitWidth":32,"isSigned":true,"name":"int"}}
```
into the following repr
```sql
{"name":"cc_call_center_sk","data_type":"Int32","nullable":true,"dict_id":0,"dict_is_ordered":false,"metadata":{}}
{"name":"cc_call_center_id","data_type":"Utf8","nullable":true,"dict_id":0,"dict_is_ordered":false,"metadata":{}}
{"name":"cc_rec_start_date","data_type":"Date32","nullable":true,"dict_id":0,"dict_is_ordered":false,"metadata":{}}
...
{"name":"RefererHash","data_type":"Int64","nullable":false,"dict_id":0,"dict_is_ordered":false,"metadata":{}}
{"name":"URLHash","data_type":"Int64","nullable":false,"dict_id":0,"dict_is_ordered":false,"metadata":{}}
{"name":"CLID","data_type":"Int32","nullable":false,"dict_id":0,"dict_is_ordered":false,"metadata":{}}
```